mirage/factories/submission.js

```diff
@@ -85,6 +85,15 @@ export default Factory.extend({
           completedAt: submission.createdAt,
         });
       }
+
+      server.create('course-leaderboard-entry', {
+        status: submission.repository.lastSubmissionIsEvaluating ? 'evaluating' : submission.repository.allStagesAreComplete ? 'completed' : 'idle',
```

Comment:
@libmartinito I have no idea how this works - `submission.repository` isn't an Ember data, model, it is a mirage model (it doesn't have all the computed properties like `allStagesAreComplete`, `lastSubmissionIsEvaluating` etc.). Can we double-check this?

Reply:
yep computed properties doesn't exist. the things you pointed out here (undefined check, updating a computed property, trying to create a leaderboard entry) originated from me thinking that course leaderboard entries aren't created. turns out, they're created in the course stage completions factory.

the problem that arose after creating a leaderboard entry here was i got one where the completedCourseStageSlugs were undefined. that's why i tried forcibly updating the value + adding an undefined check